source/common/http/http1/codec_impl.cc

```diff
@@ -532,6 +534,15 @@ int ConnectionImpl::onHeadersCompleteBase() {
       ENVOY_CONN_LOG(trace, "codec entering upgrade mode.", connection_);
       handling_upgrade_ = true;
     }
+  } else if (connection_header_sanitization_ && current_header_map_->Connection()) {
```

Comment:
can you add release notes at docs/root/intro/version_history.rst including the flag to flip to revert to prior behavior?

Reply:
Sure thing!